privacy-protections/storage-partitioning/helpers/commonTests.js

```diff
@@ -0,0 +1,552 @@
+
+            document.cookie = `jsdata=${data}; expires= Wed, 21 Aug 2030 20:00:00 UTC; Secure; SameSite=${sameSite}`;
+        },
+        retrieve: () => {
+            return document.cookie.match(/jsdata=([0-9a-z-]+)/)[1];
+        },
+        validate: validateStorageAPI
+    },
+    localStorage: {
+        type: 'storage',
+        store: (data) => {
+            localStorage.setItem('data', data);
+        },
+        retrieve: () => {
+            return localStorage.getItem('data');
+        },
+        validate: validateStorageAPI
+    },
+    sessionStorage: {
+        type: 'storage',
+        store: (data) => {
+            sessionStorage.setItem('data', data);
+        },
+        retrieve: () => {
+            return sessionStorage.getItem('data');
+        },
+        validate: validateStorageAPI
+    },
+    IndexedDB: {
+        type: 'storage',
+        store: (data) => {
+            return DB('data').then(db => Promise.all([db.deleteAll(), db.put({ id: data })])).then(() => 'OK');
+        },
+        retrieve: () => {
+            return DB('data').then(db => db.getAll()).then(data => data[0].id);
+        },
+        validate: validateStorageAPI
+    },
+    WebSQL: {
+        type: 'storage',
+        store: (data) => {
+            let res, rej;
+            const promise = new Promise((resolve, reject) => { res = resolve; rej = reject; });
+
+            const db = window.openDatabase('data', '1.0', 'data', 2 * 1024 * 1024);
+
+            db.transaction(tx => {
+                tx.executeSql('CREATE TABLE IF NOT EXISTS data (value)', [], () => {
+                    tx.executeSql('DELETE FROM data;', [], () => {
+                        tx.executeSql('INSERT INTO data (value) VALUES (?)', [data], () => res(), (sql, e) => rej('err - insert ' + e.message));
+                    }, (sql, e) => rej('err - delete ' + e.message));
+                }, (sql, e) => rej('err - create ' + e.message));
+            });
+
+            return promise;
+        },
+        retrieve: () => {
+            if (!window.openDatabase) {
+                throw new Error('Unsupported');
+            }
+            let res, rej;
+            const promise = new Promise((resolve, reject) => { res = resolve; rej = reject; });
+
+            const db = window.openDatabase('data', '1.0', 'data', 2 * 1024 * 1024);
+
+            db.transaction(tx => {
+                tx.executeSql('SELECT * FROM data', [], (tx, d) => {
+                    res(d.rows[0].value);
+                }, (sql, e) => rej('err - select ' + e.message));
+            });
+
+            return promise;
+        },
+        validate: validateStorageAPI
+    },
+    'Cache API': {
+        type: 'storage',
+        store: (data) => {
+            return caches.open('data').then((cache) => {
+                const res = new Response(data, {
+                    status: 200
+                });
+
+                return cache.put('/cache-api-response', res);
```

Comment:
let's keep it in the `/partitioning` namespace with other stuff maybe to avoid conflicts with other tests?